pkg/build/admission/defaults/admission_test.go

```diff
@@ -130,3 +130,143 @@ func TestIncrementalDefaults(t *testing.T) {
+					kapi.ResourceName(kapi.ResourceMemory): resource.MustParse("2G"),
+				},
+			},
+			ExpectedResource: kapi.ResourceRequirements{
+				Limits: kapi.ResourceList{
+					kapi.ResourceName(kapi.ResourceCPU):    resource.MustParse("20"),
+					kapi.ResourceName(kapi.ResourceMemory): resource.MustParse("2G"),
+				},
+				Requests: kapi.ResourceList{
+					kapi.ResourceName(kapi.ResourceCPU):    resource.MustParse("20"),
+					kapi.ResourceName(kapi.ResourceMemory): resource.MustParse("2G"),
+				},
+			},
+		},
+		"BuildDefaults plugin defined limits and requests ,BUILD object defined resource requests": {
+			DefaultResource: kapi.ResourceRequirements{
+				Limits: kapi.ResourceList{
+					kapi.ResourceName(kapi.ResourceCPU):    resource.MustParse("10"),
+					kapi.ResourceName(kapi.ResourceMemory): resource.MustParse("1G"),
+				},
+				Requests: kapi.ResourceList{
+					kapi.ResourceName(kapi.ResourceCPU):    resource.MustParse("10"),
+					kapi.ResourceName(kapi.ResourceMemory): resource.MustParse("1G"),
+				},
+			},
+			BuildResource: kapi.ResourceRequirements{
+				Requests: kapi.ResourceList{
+					kapi.ResourceName(kapi.ResourceCPU):    resource.MustParse("20"),
+					kapi.ResourceName(kapi.ResourceMemory): resource.MustParse("2G"),
+				},
+			},
+			ExpectedResource: kapi.ResourceRequirements{
+				Limits: kapi.ResourceList{
+					kapi.ResourceName(kapi.ResourceCPU):    resource.MustParse("10"),
+					kapi.ResourceName(kapi.ResourceMemory): resource.MustParse("1G"),
+				},
+				Requests: kapi.ResourceList{
+					kapi.ResourceName(kapi.ResourceCPU):    resource.MustParse("20"),
+					kapi.ResourceName(kapi.ResourceMemory): resource.MustParse("2G"),
+				},
+			},
+		},
+		"BuildDefaults plugin defined limits and requests ,BUILD object defined resource limits": {
+			DefaultResource: kapi.ResourceRequirements{
+				Limits: kapi.ResourceList{
+					kapi.ResourceName(kapi.ResourceCPU):    resource.MustParse("10"),
+					kapi.ResourceName(kapi.ResourceMemory): resource.MustParse("1G"),
+				},
+				Requests: kapi.ResourceList{
+					kapi.ResourceName(kapi.ResourceCPU):    resource.MustParse("10"),
+					kapi.ResourceName(kapi.ResourceMemory): resource.MustParse("1G"),
+				},
+			},
+			BuildResource: kapi.ResourceRequirements{
+				Limits: kapi.ResourceList{
+					kapi.ResourceName(kapi.ResourceCPU):    resource.MustParse("20"),
+					kapi.ResourceName(kapi.ResourceMemory): resource.MustParse("2G"),
+				},
+			},
+			ExpectedResource: kapi.ResourceRequirements{
+				Limits: kapi.ResourceList{
+					kapi.ResourceName(kapi.ResourceCPU):    resource.MustParse("20"),
+					kapi.ResourceName(kapi.ResourceMemory): resource.MustParse("2G"),
+				},
+				Requests: kapi.ResourceList{
+					kapi.ResourceName(kapi.ResourceCPU):    resource.MustParse("10"),
+					kapi.ResourceName(kapi.ResourceMemory): resource.MustParse("1G"),
+				},
+			},
+		},
+		"BuildDefaults plugin defined nothings ,BUILD object defined resource limits": {
```

Comment:
"BuildDefaults plugin defined nothing, Build object defined resource limits"

(nothing is singular, comma goes immediately after nothing, Build doesn't need to be all caps.  Same changes in the other test descriptions)